modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/plugin/ArtifactPluginDescriptorFactory.java

```diff
@@ -80,12 +91,36 @@ public ArtifactPluginDescriptor create(File pluginFolder) throws ArtifactDescrip
     }
 
     descriptor.setClassLoaderModel(classLoaderModelBuilder.build());
-
+    descriptor.setExtensionModelDescriptorProperty(lookupJsonDescriptor(pluginFolder));
     descriptor.setBundleDescriptor(createDefaultPluginBundleDescriptor(descriptor.getName()));
 
     return descriptor;
   }
 
+  private DescriptorProperty lookupJsonDescriptor(File pluginFolder) {
+    DescriptorProperty descriptorProperty = null;
+    final File mulePluginJsonFile = new File(pluginFolder, META_INF + separator + MULE_PLUGIN_JSON);
+    if (mulePluginJsonFile.exists()) {
+
+      MulePluginModel mulePluginModel = getMulePluginJsonDescriber(mulePluginJsonFile);
+      Optional<MulePluginProperty> extensionModelDescriptor = mulePluginModel.getExtensionModelLoaderDescriptor();
```

Comment:
renaming `extensionModelDescriptor` for `extensionModelLoaderDescriptor` in this class, and also making the change in mule-api from `MulePluginProperty` to `MulePluginLoaderProperty`